### ISSUE TITLE
No dependencies are detected in files using optional chaining

### ISSUE THREAD

**Issue by haggholm** [depcheck/depcheck#291]
### Bug Description

If a file contains the optional chaining proposal `object?.optionalProperty`, parsing appears to fail and *no* dependencies are detected in the file.

### Code snippets
Where the dependency is used:

```jsx
import _ from 'lodash';

export default function foo(ob) {
    return _.identity(ob?.x);
}
```

Where the dependency is listed in package.json:

The dependency is *not* listed in `package.json` (in my sample scenario, it doesn’t contain any dependencies at all).

Expected outcome:

```
Missing dependencies
* lodash
```

Actual outcome: No output/nothing detected.

### Versions
 - `node -v`: v10.4.0
 - `npm -v`: 6.1.0
 - `depcheck -v`: No output, it doesn’t seem to recognise the `-v` flag. However, the version is 0.6.11.

### Extra info

`depcheck --json`

```json
{
  "dependencies": [],
  "devDependencies": [],
  "missing": {},
  "using": {},
  "invalidFiles": {
    "/client/tmp/test.jsx": "SyntaxError: Unexpected token (4:25)\n    at Parser.pp$5.raise (/client/tmp/node_modules/babylon/lib/index.js:4454:13)\n    at Parser.pp.unexpected (/client/tmp/node_modules/babylon/lib/index.js:1761:8)\n    at Parser.pp$3.parseExprAtom (/client/tmp/node_modules/babylon/lib/index.js:3750:12)\n    at Parser.parseExprAtom (/client/tmp/node_modules/babylon/lib/index.js:7238:22)\n    at Parser.pp$3.parseExprSubscripts (/client/tmp/node_modules/babylon/lib/index.js:3494:19)\n    at Parser.pp$3.parseMaybeUnary (/client/tmp/node_modules/babylon/lib/index.js:3474:19)\n    at Parser.pp$3.parseExprOps (/client/tmp/node_modules/babylon/lib/index.js:3404:19)\n    at Parser.pp$3.parseMaybeConditional (/client/tmp/node_modules/babylon/lib/index.js:3381:19)\n    at Parser.pp$3.parseMaybeAssign (/client/tmp/node_modules/babylon/lib/index.js:3344:19)\n    at Parser.parseMaybeAssign (/client/tmp/node_modules/babylon/lib/index.js:6474:20)"
  },
  "invalidDirs": {}
}
```
I tried to add `'optionalChaining'` to the list of plugins in `depcheck/dist/jsx.js'`, but this appeared to have no effect.

**Comment by mnkhouri** [comment]
Thanks for the report @haggholm. I did a bit of digging around that plugin:

Our current parser is [Babylon@6.1.21](https://github.com/depcheck/depcheck/blob/master/package.json#L52), which does not support the `'optionalChaining'` plugin [per the documentation for 6.26.3 (so it's definitely not in our 6.1.21)](https://babeljs.io/docs/en/6.26.3/babylon#plugins)

As of Babel@7, Babylon was [renamed to @babel/parser](https://github.com/babel/babylon#babylon-has-been-moved-into-the-main-babel-mono-repo-as-babelparser), which does support the `optionalChaining` plugin, [according to documention for `@babel/parser@7.1.0`](https://babeljs.io/docs/en/babel-parser#ecmascript-proposals-https-githubcom-babel-proposals)

So the solution is for us to update `babylon` to `@babel/parser`. I'll try to get to this this weekend, but if anyone else feels like tackling it in the meantime, a PR would be greatly appreciated! I would guess it's a drop-in replacement, or minimal changes

**Comment by LinusU** [comment]
> So the solution is for us to update `babylon` to `@babel/parser`

Would be awesome to get this in 👏

**Comment by rumpl** [comment]
Fixed by #307
